src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Discovery/DiscovererCollection.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Performance.Toolkit.PluginManager.Core.Discovery
+{
+    public sealed class DiscovererCollection
+    {
+        private readonly Dictionary<Type, IPluginDiscovererSource> discoverers;
```

Comment:
This should probably be a one to many mapping. I.e. `Dictionary<Type, IEnumerable<IPluginDiscovererSource>>`

Reply:
Refactored